dhctl/pkg/config/validation.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/go-openapi/spec"
+	"sigs.k8s.io/yaml"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/operations/phases"
+)
+
+const xUnsafeExtension = "x-unsafe"
+
+// ValidateClusterSettingsFormat parses and validates cluster configuration and resources.
+// It checks the cluster configuration yamls for compliance with the yaml format and schema.
+// Non-config resources are checked only for compliance with the yaml format and the validity of apiVersion and kind fields.
+// It can be used as an imported functionality in external modules.
+func ValidateClusterSettingsFormat(settings string) error {
+	schemaStore := NewSchemaStore()
+
+	bigFileTmp := strings.TrimSpace(settings)
+	docs := regexp.MustCompile(`(?:^|\s*\n)---\s*`).Split(bigFileTmp, -1)
```

Comment:
optional. 
We use one regexp in multiple location.
We can move regexp in global variable

Reply:
Fixed!